src/bootstrap/compile.rs

```diff
@@ -312,8 +310,15 @@ pub fn std_cargo(builder: &Builder<'_>, target: TargetSelection, stage: u32, car
     // If `compiler-rt` is available ensure that the `c` feature of the
     // `compiler-builtins` crate is enabled and it's configured to learn where
     // `compiler-rt` is located.
-    let compiler_builtins_root = builder.src.join("src/llvm-project/compiler-rt");
-    let compiler_builtins_c_feature = if compiler_builtins_root.exists() {
+    let compiler_builtins_c_feature = if builder.config.optimized_compiler_builtins {
+        if !builder.is_rust_llvm(target) {
+            panic!(
+                "need a managed LLVM submodule for optimized intrinsics support; unset `llvm-config` or `optimized-compiler-builtins`"
+            );
```

Comment:
If this check is present, there has to be a way to provide the path to the `compiler-rt` source code of the custom LLVM, otherwise it will be impossible to use custom LLVMs on some platforms (see https://github.com/rust-lang/rust/pull/101833#issuecomment-1270191721).

What I would do is have a check like:

```rust
if optimized_compiler_builtins && (!is_rust_llvm() || compiler_rt_src_path)
```

Reply:
Do you mean

```rust
if optimized_compiler_builtins && !(is_rust_llvm() || compiler_rt_src_path)
```

?

Reply:
Yep, sorry!